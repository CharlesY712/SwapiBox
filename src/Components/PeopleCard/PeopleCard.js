import React from 'react';
import './PeopleCard.css';
import PropTypes from 'prop-types';

const PeopleCard = ({name, species, home, population}) => {
  return (
    <div className='card'>
      <h1>{name}</h1>
      <h4>Species: {species}</h4>
      <h4>Homeworld: {home}</h4>
      <h4>population: {population}</h4>
    </div>
  );
};

export default PeopleCard;