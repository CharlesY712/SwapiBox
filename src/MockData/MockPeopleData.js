const MockPeopleData = {
  homeworld: "Tatooine",
  name: "Luke Skywalker",
  population: "200000",
  species: "Human"
};

export default MockPeopleData;