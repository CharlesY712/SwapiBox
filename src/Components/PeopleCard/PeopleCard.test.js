import React from 'react';
import {shallow} from 'enzyme';
import PeopleCard from './PeopleCard';
import MockPeopleData from '../../MockData/MockPeopleData';

describe('PeopleCard', () => {
  const mockPeopleData = MockPeopleData;
  let wrapper = shallow(<PeopleCard 
    name={mockPeopleData.name}
    species={mockPeopleData.species}
    home={mockPeopleData.homeworld}
    population={mockPeopleData.population}/>);

  it('should exist', () => {
    expect(wrapper).toBeDefined();
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});