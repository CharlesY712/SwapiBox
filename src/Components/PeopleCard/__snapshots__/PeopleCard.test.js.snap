// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PeopleCard should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PeopleCard
    home="Tatooine"
    name="Luke Skywalker"
    population="200000"
    species="Human"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <h1>
          Luke Skywalker
</h1>,
        <h4>
          Species: 
          Human
</h4>,
        <h4>
          Homeworld: 
          Tatooine
</h4>,
        <h4>
          population: 
          200000
</h4>,
      ],
      "className": "card",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Luke Skywalker",
        },
        "ref": null,
        "rendered": "Luke Skywalker",
        "type": "h1",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Species: ",
            "Human",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Species: ",
          "Human",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Homeworld: ",
            "Tatooine",
          ],
        },
        "ref": null,
        "rendered": Array [
          "Homeworld: ",
          "Tatooine",
        ],
        "type": "h4",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "population: ",
            "200000",
          ],
        },
        "ref": null,
        "rendered": Array [
          "population: ",
          "200000",
        ],
        "type": "h4",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1>
            Luke Skywalker
</h1>,
          <h4>
            Species: 
            Human
</h4>,
          <h4>
            Homeworld: 
            Tatooine
</h4>,
          <h4>
            population: 
            200000
</h4>,
        ],
        "className": "card",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Luke Skywalker",
          },
          "ref": null,
          "rendered": "Luke Skywalker",
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Species: ",
              "Human",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Species: ",
            "Human",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Homeworld: ",
              "Tatooine",
            ],
          },
          "ref": null,
          "rendered": Array [
            "Homeworld: ",
            "Tatooine",
          ],
          "type": "h4",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "population: ",
              "200000",
            ],
          },
          "ref": null,
          "rendered": Array [
            "population: ",
            "200000",
          ],
          "type": "h4",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
